### PR TITLE
F gradient test

### PR DESCRIPTION
**Pull request by ziyiyin97**
1. Add isic option to parse_commandline in test_utils.jl
2. Fix fwiGradientTest.jl by including test_utils.jl
3. Add Taylor-series gradient test for system J(q) = 0.5||Fq-d||_2^2 for both array and judiVector, w/ options of inverse scattering imaging condition and free surface

### CODE REVIEW

File: test/fwiGradientTest.jl [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 using JUDI.TimeModeling, Test, LinearAlgebra, PyPlot, Printf
 
+include("test_utils.jl")
```
Comment (mloubout) [inline test/fwiGradientTest.jl:11]: No, this is already included in `runtest.jl` and is not needed if test is run as it's supposed to in julia.

File: test/gradient_test_F.jl [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# 2D gradient test for system 0.5*||Fq-d||_2^2
+# Single simultaneous source w/ random weights for both Array and judiVector
+# Author: Ziyi Yin, ziyi.yin@gatech.edu
+# Date: August 2020
+
+using JUDI.TimeModeling, Test, LinearAlgebra, PyPlot, Printf
+
+include("test_utils.jl")
```
Comment (mloubout) [inline test/gradient_test_F.jl:8]: Again, NO

File: test/gradient_test_F.jl [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+parsed_args = parse_commandline()
+
+println("Gradient test for system 0.5*||Fq-d||_2^2")
+
+println(parsed_args["nlayer"], " layers and tti: ",
+		parsed_args["tti"], " and freesurface: ", parsed_args["fs"],
+		" and isic: ", parsed_args["isic"] )
```
Comment (mloubout) [inline test/gradient_test_F.jl:16]: `isic` has no effect on F at all, it is the Inverse scattering imaging condition and only impact the jacobian J and its adjoint, this is unnecessary cluter.

File: test/test_utils.jl [hunk 4]
```diff
@@ -101,6 +101,9 @@ function parse_commandline()
         "--fs"
             help = "Free surface, default False"
             action = :store_true
+        "--isic"
```
Comment (mloubout) [inline test/test_utils.jl:104]: remove, unsused (since F doesn't use it)

File: test/gradient_test_F.jl [hunk 5]
```diff
@@ -0,0 +1,120 @@
+# 2D gradient test for system 0.5*||Fq-d||_2^2
```
Comment (mloubout) [inline test/gradient_test_F.jl:1]: need to add this test file to `runtest.jl`

**Comment by ziyiyin97** [comment]
I have deleted changes on test_utils.jl and add one more test to compare gradients from Array and judiVector/judiWeights

**Comment by mloubout** [comment]
ping

**Pull request closed without merging**